### PR TITLE
Added support for custom framework root

### PR DESCRIPTION
**Pull request by raptorxcz**
I wanted to create custom XCTestCase class. This class should be base for all my test cases in all my frameworks. Without this PR XCTest framework cannot be embeded into framework.

### CODE REVIEW

File: Docs/ProjectSpec.md [hunk 1]
```diff
@@ -404,6 +404,23 @@ targets:
     type: framework
 ```
 
+**SDK Dependency**
+
+- [ ] **embed**: **String** - Root of framework path. Default value is `BUILT_PRODUCTS_DIR`
```
Comment (yonaskolb) [inline Docs/ProjectSpec.md]:
```suggestion
- [ ] **root**: **String** - Root of framework path. Default value is `BUILT_PRODUCTS_DIR`
```

Comment (raptorxcz) [reply Docs/ProjectSpec.md]: I am sorry 🙈 I fixed it in commit d23271a073c736a85583840fef39543e51ae6d6b.

Comment (yonaskolb) [reply Docs/ProjectSpec.md]: Perhaps we can also mention that a common possible value for this is `DEVELOPER_DIR`. Are there other common cases?

Comment (yonaskolb) [reply Docs/ProjectSpec.md]: @raptorxcz

**Comment by yonaskolb** [comment]
Great PR @raptorxcz! 👍

File: Docs/ProjectSpec.md (hunk 1 above)
Comment (raptorxcz) [reply Docs/ProjectSpec.md]: Available value are:

```
    case none
    case absolute
    case group
    case sourceRoot
    case buildProductsDir
    case sdkRoot
    case developerDir
    case custom(String)
```

We can allow user to set any of the value without `custom` and map it directly to this enum, or we can let user to set any value he wants and just add more info to ProjectSpec.md file. What would you prefer?

Comment (yonaskolb) [reply Docs/ProjectSpec.md]: I think keeping it as custom is fine. But maybe we can just add:
```
Root of framework path, for example `DEVELOPER_DIR`. Default value is `BUILT_PRODUCTS_DIR`
```

**Comment by yonaskolb** [comment]
Could you also add a `DEVELOPER_DIR` dependency to the `TestProject` fixture

**Comment by raptorxcz** [comment]
Done 125b9ab904b1fea507acf3d34d9f7035fd15a344.

**Comment by raptorxcz** [comment]
Fixed tests

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 #### Added
 - Added support for `app-extension.intents-service` target type [#536](https://github.com/yonaskolb/XcodeGen/pull/536) @yonaskolb
+- Added support for custom framework root [#562](https://github.com/yonaskolb/XcodeGen/pull/562) @raptorxcz
```
Comment (yonaskolb) [inline CHANGELOG.md]:
```suggestion
- Added support for custom `root` in `sdk` dependency [#562](https://github.com/yonaskolb/XcodeGen/pull/562) @raptorxcz
```

**Comment by yonaskolb** [comment]
Nice! Just one minor wording update in the change log and we can merge! 😀

File: CHANGELOG.md (hunk 2 above)
Comment (raptorxcz) [reply CHANGELOG.md]: ok done in 0533dda0a6966c2459b9a6881679d38f3be256c4

**Comment by raptorxcz** [comment]
Except this PR, another PRs https://github.com/yonaskolb/XcodeGen/pull/558 and https://github.com/yonaskolb/XcodeGen/pull/550 are waiting to be merged. Do you have any schedule when PRs will be merged?

thanks! 🙂

**Pull request merged**